vectorizer.py

```diff
@@ -65,11 +67,16 @@ async def vectorize(self, text: str, config: VectorInputConfig):
                 end_index = start_index + MAX_BATCH_SIZE
 
                 tokens = self.tokenize(sentences[start_index:end_index])
+                if self.mps:
+                    tokens.to("mps")
                 if self.cuda:
                     tokens.to(self.cuda_core)
                 batch_results = self.get_batch_results(tokens, sentences[start_index:end_index])
                 batch_sum_vectors += self.pool_embedding(batch_results, tokens, config)
-            return batch_sum_vectors.detach() / num_sentences
+                if True in torch.isnan(batch_sum_vectors).tolist():
+                    raise Exception("Vector has NaN values and can not be json serialized") 
```

Comment:
Raising errors is going to crash the container, so maybe a better way would be to log the `exception` and return `error` to weaviate so the users can see the error in result of the data creation, without raising any error otherwise more objects are going to fail due to container not being live.

Reply:
this exception is caught in the try/catch of the controller